src/main/java/datastructs/CategoricalSample.java

```diff
@@ -0,0 +1,118 @@
+package datastructs;
+
+import java.util.ArrayList;
+import java.util.*;
+
+public class CategoricalSample implements ISample<String>
+{
+
+	/**
+	 * Constructor
+	 */
+	public CategoricalSample(String name, int capacity){
+
+		this.name = name;
+		this.data = new ArrayList<String>(capacity);
+	}
+
+	/**
+	 * Constructor
+	 */
+	public CategoricalSample(String name, List<String> data){
+		this.name = name;
+		this.data = data;
+		this.categories = null;
+	}
+	
+	 /**
+     * The name of the sample
+     */
+    public final String getName(){return name;}
+
+    /**
+     * Returns the size of the sample
+     */
+    public final int getsize(){return data.size();}
+
+
+	/**
+	  * Returns the unique categories
+	  */
+	public final Set<String> getCategories(){
```

Comment:
sth is wrong with the formatting

Reply:
I suggest ew use the standard formatting IntelliJ provides